### PR TITLE
test: move "dbservices" tests to Ginkgo style

### PR DESCRIPTION
**Pull request by blgm**
Requested in PR comment for https://github.com/cloudfoundry/cloud-service-broker/pull/983

[#187331719](https://www.pivotaltracker.com/story/show/187331719)

### CODE REVIEW

File: dbservice/migrations_test.go [hunk 1]
```diff
@@ -1,175 +1,84 @@
...
+		Entry("v4.1-to-v4.2", 3, nil),
+		Entry("up-to-date", numMigrations-1, nil),
+		Entry("future", numMigrations, "the database you're connected to is newer than this tool supports"),
+		Entry("far-future", math.MaxInt32, "the database you're connected to is newer than this tool supports"),
+	)
+
+	DescribeTable(
+		"RunMigrations() failures",
+		func(lastMigration int, expectedError string) {
+			db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
```
Comment (zucchinidev) [inline dbservice/migrations_test.go:39]: praise: I wasn't aware of the in-memory feature. It seems much more convenient than managing files.

File: dbservice/migrations_test.go [hunk 2]
```diff
@@ -1,175 +1,84 @@
...
+		Entry("future", numMigrations, "the database you're connected to is newer than this tool supports"),
+		Entry("far-future", math.MaxInt32, "the database you're connected to is newer than this tool supports"),
+	)
+
+	Describe("RunMigrations() behavior", func() {
+		var db *gorm.DB
+
+		BeforeEach(func() {
+			var err error
+			// The tests don't pass when using an ":memory:" database as opposed to a real file. Presumably a GORM feature.
```
Comment (zucchinidev) [inline dbservice/migrations_test.go:56]: :eyes:

File: dbservice/migrations_test.go [hunk 3]
```diff
@@ -1,175 +1,84 @@
...
-	for tn, tc := range cases {
-		t.Run(tn, func(t *testing.T) {
-			db, err := gorm.Open(sqlite.Open("test.sqlite3"), &gorm.Config{})
-			defer func() {
-				_ = os.Remove("test.sqlite3")
-			}()
-			if err != nil {
-				t.Fatal(err)
+		It("can run migrations multiple times", func() {
+			for range 10 {
```
Comment (zucchinidev) [inline dbservice/migrations_test.go:79]: praise: It's great to see you utilizing the new range feature.

File: dbservice/recover_in_progress_operations_test.go [hunk 4]
```diff
@@ -1,83 +1,60 @@
...
+
+		By("no updating other operations")
+		var r2 models.TerraformDeployment
+		Expect(db.Where("id = ?", okID).First(&r2).Error).To(Succeed())
+		Expect(r2.LastOperationState).To(Equal("succeeded"))
+		Expect(r2.LastOperationMessage).To(Equal("fake-type succeeded"))
+
+		By("logging the expected message")
+		Expect(logger.Buffer().Contents()).To(SatisfyAll(
+			ContainSubstring(`"message":"test.recover-in-progress-operations.mark-as-failed"`),
```
Comment (zucchinidev) [inline dbservice/recover_in_progress_operations_test.go:56]: praise: much more readable

**Review by zucchinidev — APPROVED** [review]

**Review by zucchinidev — APPROVED** [review]
@blgm great job! we need to fix the conflicts

**Review by pivotal-marcela-campo — APPROVED** [review]

**Pull request merged**